src/main/java/uk/gov/pay/adminusers/model/CreateUserRequest.java

```diff
@@ -42,30 +44,39 @@ public static CreateUserRequest from(String username, String password, String em
-        final List<String> gatewayAccountIds = new ArrayList<>();
-        final List<String> serviceExternalIds = new ArrayList<>();
-        try {
-            if (node.get(FIELD_GATEWAY_ACCOUNT_IDS) != null) {
-                node.get(FIELD_GATEWAY_ACCOUNT_IDS)
-                        .iterator()
-                        .forEachRemaining((gatewayAccountId) -> gatewayAccountIds.add(gatewayAccountId.asText()));
-                gatewayAccountIds.sort(numericallyThenLexicographically());
-            }
-            if (node.get(FIELD_SERVICE_EXTERNAL_IDS) != null) {
-                node.get(FIELD_SERVICE_EXTERNAL_IDS)
-                        .iterator()
-                        .forEachRemaining((serviceExternalIdNode) -> serviceExternalIds.add(serviceExternalIdNode.asText()));
-            }
-            String username = node.get(FIELD_USERNAME).asText();
-            String password = getOrElseRandom(node.get(FIELD_PASSWORD), randomUuid());
-            String email = node.get(FIELD_EMAIL).asText();
-            String telephoneNumber = node.get(FIELD_TELEPHONE_NUMBER).asText();
-            String otpKey = getOrElseRandom(node.get(FIELD_OTP_KEY), newId());
-            String features = getOrElseRandom(node.get(FIELD_FEATURES), null);
-            return from(username, password, email, gatewayAccountIds, serviceExternalIds, otpKey, telephoneNumber, features);
-        } catch (NullPointerException e) {
-            throw new RuntimeException("Error retrieving required fields for creating a user", e);
+        final List<String> gatewayAccountIds = safelyGetList(node, FIELD_GATEWAY_ACCOUNT_IDS);
+        gatewayAccountIds.sort(numericallyThenLexicographically());
+
+        final List<String> serviceExternalIds = safelyGetList(node, FIELD_SERVICE_EXTERNAL_IDS);
+
+        String username = safelyGetRequiredNodeAsText(node, FIELD_USERNAME);
+        String password = getOrElseRandom(node.get(FIELD_PASSWORD), randomUuid());
+        String email = safelyGetRequiredNodeAsText(node, FIELD_EMAIL);
+        String telephoneNumber = safelyGetRequiredNodeAsText(node, FIELD_TELEPHONE_NUMBER);
+        String otpKey = getOrElseRandom(node.get(FIELD_OTP_KEY), newId());
+        String features = getOrElseRandom(node.get(FIELD_FEATURES), null);
+        return from(username, password, email, gatewayAccountIds, serviceExternalIds, otpKey, telephoneNumber, features);
+    }
+
+    private static List<String> safelyGetList(JsonNode node, String fieldName) {
+        if (node == null || node.get(fieldName) == null) {
+            return Collections.emptyList();
         }
+
+        var results = new ArrayList<String>();
+        node.get(fieldName)
+                .iterator()
+                .forEachRemaining(nodeValue -> safelyGetNodeAsText(nodeValue).ifPresent(results::add));
+        return results;
+    }
+
+    private static String safelyGetRequiredNodeAsText(JsonNode node, String fieldName) {
```

Comment:
There’s nothing really safe about this! I’d call it `getRequiredNodeOrFail` or something.

Reply:
Yeah good point, I named the method then later moved the common safe bit out to be shared with the safelyGetList. I'll update 👍